src/blockly/generators/propc/gpio.js

```diff
@@ -1740,36 +1728,58 @@ Blockly.Blocks.sd_read = {
         this.setSdMode(mode);
     },
     setSdMode: function (mode) {
-        if(this.getInput('SIZE')) {
+        if (this.getInput('SIZE')) {
```

Comment:
This is a new addition - if a value block is attached - save that connection and restore it after the block is rebuilt.